crates/l2/prover/zkvm/interface/sp1/src/main.rs

```diff
@@ -0,0 +1,59 @@
+#![no_main]
+
+use ethrex_blockchain::{validate_block, validate_gas_used};
+use ethrex_vm::{execute_block, get_state_transitions, EvmState};
+use zkvm_interface::{
+    io::{ProgramInput, ProgramOutput},
+    trie::update_tries,
+};
+
+sp1_zkvm::entrypoint!(main);
+
+pub fn main() {
+    let ProgramInput {
+        block,
+        parent_block_header,
+        db,
+    } = sp1_zkvm::io::read::<ProgramInput>();
+
+    let mut state = EvmState::from(db.clone());
+
+    // Validate the block pre-execution
+    validate_block(&block, &parent_block_header, &state).expect("invalid block");
+
+    // Validate the initial state
+    let (mut state_trie, mut storage_tries) = db
+        .build_tries()
+        .expect("failed to build state and storage tries or state is not valid");
+
+    let initial_state_hash = state_trie.hash_no_commit();
+    if initial_state_hash != parent_block_header.state_root {
+        panic!("invalid initial state trie");
+    }
+
+    let receipts = execute_block(&block, &mut state).expect("failed to execute block");
+    validate_gas_used(&receipts, &block.header).expect("invalid gas used");
+
+    let cumulative_gas_used = match receipts.last() {
```

Comment:
Ditto.

Reply:
Done in [c7aa3b7](https://github.com/lambdaclass/ethrex/pull/1466/commits/c7aa3b7938549869411b412f53caf64eac0bbb2e)